src/modules/advancedOrders/hooks/useAdvancedOrdersActions.ts

```diff
@@ -46,10 +57,31 @@ export function useAdvancedOrdersActions() {
     console.log('On change recipient')
   }, [])
 
-  // TODO: implement this one
   const onSwitchTokens = useCallback(() => {
-    console.log('On switch tokens')
-  }, [])
+    if (!isWrapOrUnwrap) {
```

Comment:
The code is very similar with `useLimitOrdersWidgetActions`, can we unify it?

Reply:
I would suggest we keep it separate.

Reply:
Why?
It it violates the DRY principle

Reply:
@nenadV91 is good you defend your views, but it helps if you give some arguments. 

In principle, this seems a clear case of code duplication, and this leads to errors. It's very easy to fix in the future something for the limit orders logic, and forget to do the same in twap.

Reply:
There are some differences between those two

For example in twap orders we have
```javascript
      updateQuoteState({ response: null })
      updateAdvancedOrderRawState({
        inputCurrencyId: outputCurrencyId,
        outputCurrencyId: inputCurrencyId,
        inputCurrencyAmount: FractionUtils.serializeFractionToJSON(outputCurrencyAmount),
        outputCurrencyAmount: null,
      })
```

And in limit orders
```javascript
      updateLimitOrdersState({
        inputCurrencyId: outputCurrencyId,
        outputCurrencyId: inputCurrencyId,
        inputCurrencyAmount: FractionUtils.serializeFractionToJSON(outputCurrencyAmount),
        outputCurrencyAmount: FractionUtils.serializeFractionToJSON(inputCurrencyAmount),
        orderKind: orderKind === OrderKind.SELL ? OrderKind.BUY : OrderKind.SELL,
      })
```

And we can't use the same code for TWAP orders as we use for Limit orders.

Reply:
Good point. Anyway, we can unify it:
https://github.com/cowprotocol/cowswap/pull/2575

Reply:
Meged